test/com/adgoji/mollie/utils.clj

```diff
@@ -0,0 +1,73 @@
+(ns com.adgoji.mollie.utils
+  (:require
+   [clojure.edn :as edn]
+   [com.adgoji.mollie :as mollie]
+   [com.adgoji.mollie.api :as mollie.api]
+   [com.adgoji.mollie.customer :as customer]
+   [com.adgoji.mollie.payment :as payment]
+   [com.adgoji.mollie.subscription :as subscription]
+   [com.adgoji.mollie.mandate :as mandate]))
+
+(def ^:private secrets-file "/.adgoji-mollie-secrets.edn")
+(def ^:private secrets
+  (try
+    (-> (str (System/getenv "HOME") secrets-file)
+        slurp
+        edn/read-string)
+    (catch Exception _ {})))
+
+(def api-key
+  (or (System/getenv "MOLLIE_API_KEY")
+      (:api-key secrets)))
+
+(def profile-id
+  (or (System/getenv "MOLLIE_PROFILE_ID")
+      (:profile-id secrets)))
+
+(def test-purpose "api-client-test")
+
+(def default-metadata {:purpose test-purpose})
+
+(defn new-client []
+  (mollie.api/new-client {:api-key           api-key
+                          :check-response?   true
+                          :throw-exceptions? true}))
+
+(defn cleanup-payments [mollie-client customer-id]
+  (doseq [{payment-id ::payment/id}
+          (->> (mollie.api/get-payments-list mollie-client customer-id {})
+               (::mollie/payments)
+               (sequence (filter ::payment/is-cancelable)))]
+    (mollie.api/cancel-payment-by-id mollie-client payment-id)))
+
+(defn cleanup-subscriptions [mollie-client customer-id]
+  (doseq [{subscription-id ::subscription/id}
+          (->> (mollie.api/get-subscriptions-list mollie-client customer-id {})
+               (::mollie/subscriptions)
+               (sequence (remove (fn [sub]
+                                   (= (::subscription/status sub) :canceled)))))]
+    (mollie.api/cancel-subscription-by-id mollie-client customer-id subscription-id)))
+
+(defn cleanup-mandates [mollie-client customer-id]
+  (doseq [{mandate-id ::mandate/id}
+          (->> (mollie.api/get-mandates-list mollie-client customer-id {})
+               (::mollie/mandates))]
+    (mollie.api/revoke-mandate-by-id mollie-client customer-id mandate-id)))
+
+(defn- customers-to-cleaunup [client]
+  (->> (mollie.api/get-customers-list client {})
+       (::mollie/customers)
+       (sequence (filter (fn [customer]
+                           (= test-purpose (get-in customer [::customer/metadata :purpose])))))))
+(defn mollie-cleanup [client]
+  (doseq [{customer-id ::customer/id} (customers-to-cleaunup client)]
+    (cleanup-payments client customer-id)
+    (cleanup-subscriptions client customer-id)
+    (cleanup-mandates client customer-id)
+    (mollie.api/delete-customer-by-id client customer-id)))
+
+(defn system-fixture [f]
+  (try
+    (f)
+    (finally
+      (mollie-cleanup (new-client)))))
```

Comment:
Just an idea: since we use fixture now it would be nice to re-use the same client instance using dynamic variable and `binding` (you can find an example [here](https://github.com/millmobile/adgoji-google-clj/blob/0e17adce08214a315e7293f361cddf7118a3dcb2/projects/google-displayvideo-clj/test/com/adgoji/google_displayvideo_clj/fixtures.clj#L6-L12). 

Feel free to ignore this comment.

Reply:
Great idea, added